mmj2jar/PATutorial/Page503.mmp

```diff
@@ -1,71 +1,33 @@
-  "re" for real numbers, "1" for the number 1, "gt" for "greater than",
-  "le" for less than, "an" for and, and so on.
-  Most symbols are defined by an assertion named "df-NAME", where
-  NAME is the label fragment used. Note that in set.mm, "natural number"
-  means an integer that is one or larger.
-  Thus, in set.mm, "nnre" represents "the natural numbers are real numbers",
-  and "nnge1" represents "the natural numbers are greater than or equal
-  to 1".
-
-  Parentheses must be placed in specific places.
-  When a function that takes two classes and produces a class is applied
-  as part of an infix expression, the expression is always surrounded by
-  parentheses, e.g., the use of "+" in "( 2 + 2 )".  Similarly,
-  predicate expressions in infix form that take two or three wffs
-  (a true or false value) and produce a wff are also always
-  surrounded by parentheses, such as "( ph -> ps )".  In contrast,
-  a binary relation (which compares two classes and produces a wff)
-  applied in an infix expression is not surrounded by parentheses. This
-  includes set membership, for example, "1 e. RR" has no parentheses.
-
-  For more details, see the set.mm conventions section off the
-  main Metamath Proof Explorer (set.mm) home page:
-  http://us.metamath.org/mpegif/conventions.html
-
-
-* A positive integer is greater than one iff it is not equal to one.
-  (Contributed by NM, 7-Oct-2004.)
-
-50::nnre           |- ( A e. NN -> A e. RR )
-51::nnge1          |- ( A e. NN -> 1 <_ A )
-52::1re            |- 1 e. RR
-53::leltne         |- (  ( 1 e. RR /\ A e. RR /\ 1 <_ A )
-                      -> ( 1 < A <-> A =/= 1 ) )
-54:52,53:mp3an1    |- ( ( A e. RR /\ 1 <_ A ) -> ( 1 < A <-> A =/= 1 ) )
-qed:50,51,54:syl2anc
-                   |- ( A e. NN -> ( 1 < A <-> A =/= 1 ) )
-
-* Please go on to Page503.mmp.
+ Let's reuse the proof we were doing in tutorial chapter 3,
+ but remove its unnecessary third hypothesis.
+ We'll keep all the statements, but remove all of the Refs and Hyps.
+ Instead, we're going to depend purely on the advanced mmj2 unification
+ mechanisms by putting a '!' in front of all the non-hypothesis steps.
+
+ Press control-U!
+
+h              |- ( ph -> ps )
+h              |- ( ps -> ch )
+!              |- ( ph -> ( ps -> ch ) )
+!              |- ( ( ph -> ps ) -> ( ph -> ch ) )
+!qed           |- ( ph -> ch )
+
+* Notice that mmj2 was able to complete this proof
+  without any additional help, because mmj2 had the specific
+  statements it was suppposed to use and the '!' allowed it
+  to use more advanced automation to figure out the rest.
+  In this case, mmj2 was able to examine each non-hypothesis step
+  and find a combination of the given statement, previous proven steps,
+  and a Ref (theorem or axiom) that would together
+  create that step's statement. Of course, mmj2 can't always complete
+  a proof with just some statements, even with its automation, but
+  mmj2 *can* provide some help.
```

Comment:
Actually mmj2 is "complete" in the sense that if all the steps that should appear in the proof are in the worksheet, then it will find the proof, because the ! will find arbitrary one-step proofs derived from previous steps. If there are two or more steps involved, mmj2 may miss it.

Reply:
I knew that, but I think you're right, the text should be clearer here about that.

I'll try to tweak this pull request soon.

Reply:
Okay, I've changed the text (using some of your explanation, because I thought it was a good one). Hopefully that makes it clearer.